src/theory/arith/operator_elim.h

```diff
@@ -17,43 +17,26 @@
 #include <map>
 
 #include "expr/node.h"
+#include "expr/term_conversion_proof_generator.h"
+#include "theory/eager_proof_generator.h"
 #include "theory/logic_info.h"
 
 namespace CVC4 {
 namespace theory {
 namespace arith {
 
-class OperatorElim
+class OperatorElim : public EagerProofGenerator
 {
  public:
-  OperatorElim(const LogicInfo& info);
+  OperatorElim(ProofNodeManager* pnm, const LogicInfo& info);
   ~OperatorElim() {}
+  /** eliminate */
```

Comment:
I think it would be good to have some more documentation for this, since this is now the public interface.

```suggestion
  /** Eliminate operators in this term.
    *
    * Eliminate operators in term n. If n has top symbol that is not a core
    * one (including division, int division, mod, to_int, is_int, syntactic sugar
    * transcendental functions), then we replace it by a form that eliminates
    * that operator. This may involve the introduction of witness terms.
    */
```